sw/device/silicon_creator/lib/boot_data.c

```diff
@@ -665,3 +666,16 @@ rom_error_t boot_data_check(const boot_data_t *boot_data) {
 
   return kErrorBootDataInvalid;
 }
+
+rom_error_t boot_data_as_v2(boot_data_t *boot_data) {
+  switch (launder32(boot_data->version)) {
```

Comment:
nit: `version` is laundered but I don't see any hardened checks in the `case`s below. Can you add `HARDENED_CHECK_EQ(boot_data->version, kBootDataVersionN)`?